keps/sig-cloud-provider/1771-versioning-policy-for-external-cloud-providers/README.md

```diff
@@ -0,0 +1,121 @@
+Foo should support Kubernetes clusters on v1.18.X.
+
+## Motivation
+
+* improve discovery of version supportability across Kubernetes and external cloud providers.
+* promote using the latest cloud-controller-manager library based on the Kubernetes version of the cluster
+* prevent wide version skew of Kubernetes control plane and cloud-controller-manager.
+* reduce chances of incompatibility issues that occur when a cloud-controller-manager vendoring a really
+old version of Kubernetes is running on a much newer version of Kubernetes.
+
+### Goals
+
+* standardize the semantic versioning of all external cloud providers
+* reduce chances of incompatibility issues due to version skew between cloud-controller-manager and Kubernetes.
+
+### Non-Goals
+
+* improving the build/release process for publishing arifacts like binaries and docker images.
+
+## Proposal
+
+Semantic versioning used by external cloud providers should track the major and minor releases of the supported Kubernetes
+version in that release. Patch releases are not required to match. For example, release v1.18.X of the cloud-controller-manager
+for provider Foo should support Kubernetes v1.18.X. The benefits of this versioning policy include but are not limited to:
+
+* the desired version of the cloud-controller-manager is explicit based on the Kubernetes version of the cluster
+* resolves incompatibility issues occurring wbsfrom vendoring a really old version of k8s.io/kubernetes
+* ensures the vendored version of the cloud-controller-manager is up-to-date
+
+### Risks and Mitigations
+
+* timeline for feature additions are tied to the Kubernetes release cycle (assuming features are only added in minor releases).
+* maintainers of external providers will likely have to manage release branches for each minor release of Kubernetes.
+
+## Design Details
+
+Updates to release version for external cloud providers:
+* match major and minor versions of releases based on the version of k8s.io/kubernetes that is vendored to
+build the cloud-controller-manager
+* use patch releases for bug fixes pertaining to specific Kubernetes releases. In many cases bugs will be backported to
+all minor versions that Kubernetes supports.
+* release a new minor version after every new Kubernetes minor release
+
+### Test Plan
+
+Not applicable.
+
+### Graduation Criteria
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Not applicable.
+
+### Version Skew Strategy
+
+The proposed semantic versioning policy ould simplify version skew strategy for the cloud-controller-manager since
+users can treat versioning the cloud-controller-manager as part of their control plane (if they weren't arleady).
+
+## Implementation History
+
+- 2020-04-29: the KEP is accepted in the `implementable` state.
+
+## Drawbacks
+
+One motivation for external cloud providers was so to decouple the release cadence for providers from the Kubernetes release.
```

Comment:
I believe it still decouples the release cadence. I as cloud provider do not have to ship my release when K8s ships theirs. In addition I am free to generate as many patch release, as quickly as I like on top of a K8s release. I believe that was the primary cadence decoupling benefit we were looking for. So if I find an issue in *my* code (whether CSI, cloud provider implementation, my CCM offering or deployment system, I can fix any of those without forcing a new K8s release.)

Reply:
I agree that it's still a big improvement, but if we are generally saying that features are only included in minor releases, it's a step back from being able to release minor versions whenever you want.